src/hotspot/share/gc/shenandoah/shenandoahSATBMarkQueueSet.hpp

```diff
@@ -31,14 +31,15 @@
 #include "runtime/thread.hpp"
 
 class ShenandoahSATBMarkQueueSet : public SATBMarkQueueSet {
-protected:
-  virtual bool should_enqueue_buffer(SATBMarkQueue& queue);
-
+private:
+  volatile int _enqueued_count;
```

Comment:
I have a suspicion that `int` would overflow at some point in the long-running application. `size_t` would fit better, but then see the other comment that `SATBMQ` already tracks it itself.

Reply:
Right. (I only ever compare before != after, so overflow would be ok, but it doesn't matter b/c I'll remove it)